src/shared/conf-parser.c

```diff
@@ -793,6 +793,32 @@ bool stats_by_path_equal(Hashmap *a, Hashmap *b) {
         return true;
 }
 
+int config_section_parse(
+                const ConfigSectionParser *parsers,
+                size_t n_parsers,
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *userdata) {
+
+        assert(parsers);
+        assert(n_parsers > 0);
+        assert(ltype >= 0);
+        assert((size_t) ltype < n_parsers);
+        assert(userdata);
+
```

Comment:
might be good having a comment here explaining the situation. i.e. that this is a helper to use for cases where each [] configuration section shall introduce an additional object to synthesize